modules/gdscript/gdscript_compiler.cpp

```diff
@@ -1522,8 +1522,18 @@ Error GDScriptCompiler::_parse_block(CodeGen &codegen, const GDScriptParser::Blo
 				if (ret2 < 0)
 					return ERR_PARSE_ERROR;
 
+				int message_ret = 0;
+				if (as->message) {
+					message_ret = _parse_expression(codegen, as->message, p_stack_level + 1, false);
+					if (message_ret < 0)
+						return ERR_PARSE_ERROR;
+				}
+
 				codegen.opcodes.push_back(GDScriptFunction::OPCODE_ASSERT);
 				codegen.opcodes.push_back(ret2);
+				if (as->message) {
+					codegen.opcodes.push_back(message_ret);
+				}
```

Comment:
I'm pretty sure there should be an else here, which at least pushes a zero or something. Otherwise, the instruction pointer will be misaligned after message-less asserts.

Reply:
If we initialize `message_ret` as 0 there's no need for branching, just push `message_ret` regardless. It will either be 0 or some value from `_parse_expression`

Reply:
Maybe it's better to push a `NULL` to the stack instead of the address 0?

Reply:
If I use `NULL` I get:

> /Users/mitch/dev/godot-source/modules/gdscript/gdscript_compiler.cpp:1536: warning: implicit conversion of NULL constant to 'const int' [-Wnull-conversion]
>                                         codegen.opcodes.push_back(NULL);
>                                                         ~~~~~~~~~ ^~~~
>                                                                   0

Reply:
I'd still argue it will be more readable if you write it that way:


				int message_ret = 0;
				if (as->message) {
					message_ret = _parse_expression(codegen, as->message, p_stack_level + 1, false);
					if (message_ret < 0)
						return ERR_PARSE_ERROR;
				}

				codegen.opcodes.push_back(GDScriptFunction::OPCODE_ASSERT);
				codegen.opcodes.push_back(ret2);
				codegen.opcodes.push_back(message_ret);

Reply:
Sorry @wiped1, I didn't mean to ignore your comment, I just somehow managed to gloss over it. Thanks for being patient. :) The latest patch addresses this.